### PR TITLE
Die cwd die

### PR DESCRIPTION
**Pull request by rmuir**
See #10877 for reference.

Currently some code looks in CWD and places like $user.home for configuration files in a confusing way, in some cases even overriding other paths (so if you have a stray elasticsearch.yml there, good luck). 

Tests hide the problem because they add too many permissions. We fixed that here and then all tests were failing, and made hacky changes to get tests passing again.

So some of the changes here can be cleaned up / solved differently. For example we could look for the custom hunspell data directory in Environment, grant access to it, and deprecate it instead of throwing an error. But, with what is here all tests pass so I think we know enough to discuss so we can move forward.

### CODE REVIEW

File: src/main/java/org/elasticsearch/env/Environment.java [hunk 1]
```diff
@@ -69,16 +68,12 @@
...
-        this(EMPTY_SETTINGS);
-    }
-
     public Environment(Settings settings) {
         this.settings = settings;
         if (settings.get("path.home") != null) {
             homeFile = PathUtils.get(cleanPath(settings.get("path.home")));
         } else {
-            homeFile = PathUtils.get(System.getProperty("user.dir"));
+            throw new IllegalStateException("path.home is not configured");
```
Comment (tlrx) [inline src/main/java/org/elasticsearch/env/Environment.java]: :+1:

**Comment by rmuir** [comment]
if no one has opinions then, I will push in 24 hours.

**Comment by s1monw** [comment]
this looks great  - the only thing that I'd like to see is a note in the migration guide that this has been fixed and that for instance the dictionary files should be located in a different place. It's in ` docs/reference/migration/migrate_2_0.asciidoc `

**Comment by s1monw** [comment]
LGTM

**Pull request merged**